esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -3162,97 +3107,60 @@ bool CFileSprayEx::onDropZoneFiles(IEspContext &context, IEspDropZoneFilesReques
+            dropZoneInfo.getName(dropZoneName);
+            dropZoneInfo.getDirectory(directory);
+            dropZoneInfo.getComputerName(computerName); //legacy env
+            if (!dropZoneName.length() || !directory.length())
+                continue;
 
-                Owned<IConstMachineInfo> machine;
-                if (strcmp(pszNetAddr, "."))
-                    machine.setown(m_constEnv->getMachineByAddress(sNetAddr.str()));
-                else
-                {
-                    machine.setown(m_constEnv->getMachineByAddress(pszNetAddr));
-                    if (!machine)
-                        machine.setown(m_constEnv->getMachineByAddress(sNetAddr.str()));
-                }
+            bool isLinux = getPathSepChar(directory.str()) == '/' ? true : false;
+            Owned<IConstDropZoneServerInfoIterator> dropZoneServerItr = dropZoneInfo.getServers();
+            ForEach(*dropZoneServerItr)
+            {
+                IConstDropZoneServerInfo& dropZoneServer = dropZoneServerItr->query();
 
-                StringBuffer dir;
-                pDropZone.getProp("@directory", dir);
+                StringBuffer name, server, networkAddress;
+                dropZoneServer.getName(name);
+                dropZoneServer.getServer(server);
+                if (name.isEmpty() || server.isEmpty())
+                    continue;
 
-                Owned<IEspDropZone> aDropZone= createDropZone("","");
+                IpAddress ipAddr;
+                ipAddr.ipset(server.str());
+                ipAddr.getIpText(networkAddress);
 
-                if (machine)
-                {
-                    if (machine->getOS() == MachineOsLinux || machine->getOS() == MachineOsSolaris)
-                    {
-                        dir.replace('\\', '/');//replace all '\\' by '/'
-                        aDropZone->setLinux("true");
-                        osStr = "1";
-                    }
-                    else
-                    {
-                        dir.replace('/', '\\');
-                        dir.replace('$', ':');
-                        osStr = "0";
-                    }
-                }
+                Owned<IEspDropZone> aDropZone= createDropZone();
+                aDropZone->setName(dropZoneName.str());
+                aDropZone->setComputer(name.str());
+                aDropZone->setNetAddress(networkAddress.str());
 
-                aDropZone->setComputer(pszComputer);
-                aDropZone->setPath(dir.str());
-                aDropZone->setName(pszName);
-                aDropZone->setNetAddress(sNetAddr.str());
+                aDropZone->setPath(directory.str());
+                if (isLinux)
+                    aDropZone->setLinux("true");
+                if (!isEmptyString(netAddress) && ipAddr.ipequals(ipToMatch))
+                    filesFromALinux = isLinux;
 
                 dropZoneList.append(*aDropZone.getClear());
```

Comment:
Ok, it's not clear from method name that that is the intent.
Adding a comment somewhere saying that it returns all dropzones and filtered list of files would be good.